client/src/components/PostList/PostList.component.jsx

```diff
@@ -18,6 +19,55 @@ const PostList = ({ posts, defaultText, alertIfMoreThanDays, showViews }) => {
     { enabled: !!agencyShortName },
   )
   defaultText = defaultText ?? 'There are no posts to display.'
+
+  const maxScroll = useRef(0)
+
+  const googleAnalytics = useGoogleAnalytics()
+  const updateMaxScroll = () => {
+    maxScroll.current = Math.max(maxScroll.current, window.pageYOffset)
+  }
+
+  const sendMaxScrollToAnalytics = (maxScrollPossible) => {
+    // The following is required to restrict the percentage to 100 due to browser compatibility issues:
+    // https://stackoverflow.com/questions/17688595/finding-the-maximum-scroll-position-of-a-page
+    maxScrollPossible = Math.max(maxScrollPossible, maxScroll.current)
+    const maxScrollPercentage = Math.floor(
+      (maxScroll.current / maxScrollPossible) * 100,
+    )
+    googleAnalytics.sendUserEvent(
+      googleAnalytics.GA_USER_EVENTS.BROWSE,
+      agencyShortName,
+      maxScrollPercentage,
+    )
+    FullStory.event(googleAnalytics.GA_USER_EVENTS.BROWSE, {
+      agencyName_str: agencyShortName,
+      maxScroll_int: maxScrollPercentage,
+    })
+  }
+
+  const resetMaxScrollAndAddListener = () => {
+    window.addEventListener('scroll', updateMaxScroll)
+  }
+  const removeListenerAndSendEvent = (maxScrollPossible) => {
+    window.removeEventListener('scroll', updateMaxScroll)
+    if (maxScroll.current !== 0) sendMaxScrollToAnalytics(maxScrollPossible)
+  }
+
+  useEffect(() => {
+    resetMaxScrollAndAddListener()
+    // Due to browser compatibility issues, there is no single source of truth for maximum scroll height:
+    // https://stackoverflow.com/questions/17688595/finding-the-maximum-scroll-position-of-a-page
+    const pageHeight = Math.max(
+      document.body.scrollHeight,
+      document.body.offsetHeight,
+      document.documentElement.clientHeight,
+      document.documentElement.scrollHeight,
+      document.documentElement.offsetHeight,
+    )
+    const maxScrollPossible = pageHeight - window.innerHeight
+    return () => removeListenerAndSendEvent(maxScrollPossible)
+  }, [])
```

Comment:
Some commentary of what this does would be helpful. It's not immediately obvious that the hook installs a callback that gets triggered only when the component unloads